scala/src/com/google/idea/blaze/scala/run/producers/ScalaTestContextProvider.java

```diff
@@ -59,13 +64,23 @@ public RunConfigurationContext getTestContext(ConfigurationContext context) {
     ListenableFuture<TargetInfo> target =
         TestTargetHeuristic.targetFutureForPsiElement(
             testClass, TestSizeFinder.getTestSize(testClass));
+    TestContext.BlazeFlagsModification flags = null;
     if (target == null) {
       return null;
+    } else {
+      try {
+        if(target.get(2, TimeUnit.SECONDS).kindString.equals("scala_junit_test")) {
+          flags = TestContext.BlazeFlagsModification.testFilter(ProducerUtils.getTestFilterForClass(testClass));
```

Comment:
Nitpicking: I think this could be structured in a slightly more straightforward way as

```
TestContext.BlazeFlagsModification flags = getFlags();

private TestContext.BlazeFlagsModification getFlags() {
try {
  if (is junit) {
    return TestContext.BlazeFlagsModification.testFilter(ProducerUtils.getTestFilterForClass(testClass));
  }
} catch {
  //ignore
}

return new ScalatestTestSelectorFlagsModification(testClass.getQualifiedName(), classAndTestNames.testNames);
}
```

so you don't have to figure out which flags to use based on null a few lines further down.